### PR TITLE
Russian oauth providers

### PR DESCRIPTION
**Pull request by Compilerrr**
www.vk.com and www.yandex.ru providers added
If these are ok, i'll add odnoklassniki.ru and mail.ru later.

### CODE REVIEW

File: libraries/Provider/Vkontakte.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * Vkontakte OAuth2 Provider
+ *
+ * @package    CodeIgniter/OAuth2
+ * @category   Provider
+ * @author     Lavr Lyndin
+ */
+
+class OAuth2_Provider_Vkontakte extends OAuth2_Provider
+{
+	public function __construct(array $options = array())
```
Comment (philsturgeon) [inline libraries/Provider/Vkontakte.php]: Can you remove this? This should come from config like the rest or be provided inline when used, we dont want hardcoded values on providers.

**Comment by Compilerrr** [comment]
Since for one application I use only one client/secret id per provider i think it's nice place for default config. Otherwise you need some code to choose right client/secret id for specific provider.
But in general, i agree with you, it looks not good :)

**Comment by philsturgeon** [comment]
That will mess up how all code built around the provider is done though. Make it work like every other provider or I cannot merge this!

**Comment by Compilerrr** [comment]
I thought i've done it... I'm new to git maybe i'm doing something wrong?
I made a commit in my fork, is it enough?

**Pull request closed without merging**